components/ai-character.tsx

```diff
@@ -95,10 +98,23 @@ export const AICharacter = ({ form, hasInstanceAccess, save }: AIFormProps) => {
   const [categories, setCategories] = useState<Category[]>([]);
   const [generatingAll, setGeneratingAll] = useState(false);
   const [loadingMessage, setLoadingMessage] = useState(loadingMessages[0]);
+  const [selectedValues, setSelectedValues] = useState<any[]>([]);
+  const clerk = useClerk();
+
+  let role = "User";
+  if (clerk.user?.publicMetadata && clerk.user?.publicMetadata.superuser) {
+    role = "Superuser";
+  }
```

Comment:
You can use the `hasInstanceAccess` variable instead of checking the metadata. 

We already evaluate if the user has instance-level permission in the [parent page](https://github.com/nicolasdesmarais/ai-companion/blob/main/app/(root)/(routes)/ai/%5BaiId%5D/edit/%5B%5B...section%5D%5D/page.tsx#L47-L52
) and pass that to the component.